internal/archive/archive.go

```diff
@@ -86,6 +94,22 @@ func (a *ubuntuArchive) Exists(pkg string) bool {
 	return err == nil
 }
 
+type pkgInfo struct{ control.Section }
+
+var _ PackageInfo = pkgInfo{}
+
+func (info pkgInfo) Name() string    { return info.Get("Package") }
+func (info pkgInfo) Version() string { return info.Get("Version") }
+func (info pkgInfo) Arch() string    { return info.Get("Architecture") }
+func (info pkgInfo) SHA256() string  { return info.Get("SHA256") }
+
+func (a *ubuntuArchive) Info(pkg string) PackageInfo {
+	if section, _, _ := a.selectPackage(pkg); section != nil {
```

Comment:
Why drop errors on the floor?

Reply:
I just felt that the `Info()` method (and `selectPackage()` too) resembles a map read and since it's guaranteed to not hold nil values and will not fail in any expected way (as `selectPackage` only returns error when it doesn't find an entry in the map), it can just return nil on missing entry.

I understand that this is an interface and so it should be designed for unknown future changes to the implementation, but I assumed that `Archive` instances are expected to be fully loaded with archive metadata after creation. If e.g. we introduced some lazy loading in future, then even `Exists()` would have to return error in case lazy loading fails.

Anyway, after #80 gets merged, I'm going to update to return err from `selectPackage`. Python's dictionaries also fail on `KeyError` :shrug: